TRT-635: Add HGA regression tests for format.

owenlittlejohns opened the pull request:
## Description

This PR adds regression tests to HGA for:

* No format specified (should default to GeoTIFF).
* netCDF4 format specified (should output netCDF4).

To make this run, you'll need [this branch](https://github.com/nasa/harmony/pull/813).

## Jira Issue ID

TRT-635 - a catch-all for HGA sprucing up.

## Local Test Steps

* Use [this branch](https://github.com/nasa/harmony/pull/813) of Harmony.
* Build HGA locally (`./bin/build-image` in that repo)
* Start Harmony in a Box, with `LOCALLY_DEPLOYED_SERVICES=harmony-gdal-adapter`
* Run this notebook in a browser.

## PR Acceptance Checklist
* ~~Acceptance criteria met~~
* [x] Tests added/updated (if needed) and passing
* ~~Documentation updated (if needed)~~
* [x] CHANGELOG updated with the changes for this PR
* [x] Service's `version.txt` file changed if appropriate
* ~~Original file is uploaded to S3 if references are changed~~

flamingbear reviewed (commented):
I verified this while I was testing the HGA PR.  This is a nice update. Is there anything that could be done for this?

```
**Note: earthdata-hashdiff comparison is disabled because it requires Python 3.11, and that is incompatible with the version of GDAL used for GeoTIFF comparisons. This test is retained to ensure the request runs without error.**
```

any desire to add labels to the requests? I'd add 
labels = ['hga-rtest', 'hga-rtest-x'] to each request.  Although your test case numbers include more info.  they might get long if you made them 'hga-rtest-avnir-1' but might not.

owenlittlejohns commented:
Okay - so to save issues with GDAL compatibility, and GDAL reading a downloaded file differently within a container versus on a local machine, I cut over entirely to use `earthdata-hashdiff`. [You don't have to solve GDAL dependency issues if you don't have a GDAL dependency](https://imgflip.com/memegenerator/94583182/You-dont-have-to-worry)!

That's made quite a big diff in the notebook. And also supporting files. I've removed all the old actual reference files and placed them (and the missing ones) in S3 in `harmony-uat-regression-tests/hga/reference_files/1.0.0`.

I also added labels as suggested - that's a good call!

To test:

1) You can run this notebook _mostly_ against SIT or UAT within a container. Right now it will fail on the default format test for AVNIR (AVNIR test case 10), until [this PR against Harmony is merged and deployed](https://github.com/nasa/harmony/pull/813).
2) You can also run this notebook locally against your Harmony in a Box with the version of Harmony from that PR, as previously described in the test instructions. Once you've created a new conda environment (`conda env create -f hga/environment.yml`), you should be able to just run all cells and they should all pass.

owenlittlejohns commented on test/hga/reference_images/avnir_reference_eleven.json:
I think there's something wonky with the netCDF4 file produced here (potentially missing 1-D dimension variables). It feels like tackling that issue is out of scope of this PR, but definitely something to look into.

flamingbear commented:
> You can run this notebook mostly against SIT or UAT within a container. Right now it will fail on the default format test for AVNIR (AVNIR test case 10), until https://github.com/nasa/harmony/pull/813.

I'm getting an error in `UAVSAR test case 10:` also. (in UAT) 
```
AssertionError: uavsar_test_ten[/8069077_djNEON_00101_17095_012_170910_L090_CX_01_pauli_Band1_subsetted.nc](http://localhost:8890/8069077_djNEON_00101_17095_012_170910_L090_CX_01_pauli_Band1_subsetted.nc) has an unexpected set of variable
```
This doesn't seem expected but also in a netcdf output, so maybe we have a version issue somewhere.

flamingbear reviewed (changes requested):
my review seems to have gotten split between comments and review comments. 

I think this looks good with two things to address the `**notes` and the artefact cleanup.  Otherwise looks good to me.
